src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,151 @@
+      return;
+    }
+    this.hasBindingsBuff = false;
+    debug && console.log("Buff Removed @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    if (this.buffUsed === false) {
+      this.wastedProcs += 1;
+      debug && console.log("Buff Expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.usedProcs += 1;
+    }
+  }
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+    this.beginCastTimestamp = event.timestamp;
+    this.beginCastFound = true;
+
+    debug && console.log("Pyroblast Begin Cast @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id || this.beginCastFound === false) {
+      return;
+    }
+    this.beginCastFound = false;
+    debug && console.log("Pyroblast Casted @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    this.castTimestamp = event.timestamp;
+    const castTime = this.castTimestamp - this.beginCastTimestamp;
+    if (castTime >= CAST_BUFFER && this.hasBindingsBuff === true) {
+      this.isBuffed = true;
+      this.buffUsed = true;
+      debug && console.log("Buff Used @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+      if (this.isBuffed === true) {
+        this.damage += getDamageBonus(event, BONUS_DAMAGE);
+        this.isBuffed = false;
+        debug && console.log("Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      } else {
+        debug && console.log("Non Buffed Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      }
+  }
+
+  get avgBonusDamage() {
+    return this.damage / this.usedProcs;
+  }
+
+  on_finished() {
+    if (this.combatants.selected.hasBuff(SPELLS.KAELTHAS_ULTIMATE_ABILITY.id)) {
+      const adjustedFightEnding = this.owner.currentTimestamp - 5000;
+      if (this.buffAppliedTimestamp < adjustedFightEnding) {
+        this.wastedProcs += 1;
+        debug && console.log("Fight Ended with Unused Proc @ " + formatMilliseconds(this.owner.currentTimestamp - this.owner.fight.start_time));
+      } else {
+        this.totalProcs -= 1;
+      }
+    }
+  }
+
+  item() {
+    return {
+      item: ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING,
+      result: (
+        <dfn data-tip={`${this.totalProcs} Total Procs
+        <ul>
+          <li>${this.usedProcs} Procs Used</li>
+          <li>${this.wastedProcs} Procs Wasted</li>
+          <li>${formatNumber(this.avgBonusDamage)} Average Bonus Damage</li>
+        </ul>
+        `}>
+          {this.owner.formatItemDamageDone(this.damage)}
```

Comment:
It's very misleading to include the bonus damage from proc pyros as being the damage boost due to the legendary, because without the legendary players do not hardcast Pyroblast. The actual damage boost due to the bindings is the "damage increase vs. replacement", e.g. the actual damage from casting a proc pyro *minus the damage the player would have done if they'd used those 3.5 seconds to do their normal rotation*.

Furthermore, this doesn't account for Ignite.

Given the inherit inaccuracy of this number, I think it shouldn't go live displayed as is. I'm open to brainstorming possible solutions.

Reply:
i mean, its not really possible to know how much more damage it did vs casting Fireballs instead ... so im not exactly sure how i would make it more accurate in that sense. 

Maybe i could figure out how to account for Ignite, but idk if i could reasonably figure that out either.

Reply:
If "it's not really possible to calculate", then don't display a damage boost number. Show PPM or something instead.

Basically, it's better to show nothing than to show a number that you know is wrong.